Watchman.Discord/Areas/Protection/Services/UserMessagesCountService.cs

```diff
@@ -21,24 +22,45 @@ public UserMessagesCountService(IQueryBus queryBus)
 
         public int CountMessages(Contexts contexts)
         {
-            if (_serverMessagesQuantity == null || _lastUpdated.AddHours(12) < DateTime.UtcNow)
+            if (ShouldReloadCache())
             {
-                _lastUpdated = DateTime.UtcNow;
+                ReloadCache();
+            }
+
+            return GetOneUserFromOneServerCount(contexts.User.Id, contexts.Server.Id);
+        }
 
-                var query = new GetMessagesQuery();
-                var messages = _queryBus.Execute(query).Messages.ToList();
+        private bool ShouldReloadCache()
+        {
+            return _everyServersMessagesQuantity == null || _lastUpdated.AddHours(12) < DateTime.UtcNow;
+        }
 
-                var serversMessages = messages
-                    .GroupBy(x => x.Server.Id)
-                    .ToList();
+        private void ReloadCache()
+        {
+            _lastUpdated = DateTime.UtcNow;
```

Comment:
myśle że to powinno być na końcu metody a nie na początku 
chcemy czas ostatniej aktualizacji a nie informacje kiedy ostatnia była uruchomiona